src/components/CollectionsAndPalettes/common/CollectionsToolbar/CollectionsToolbar.tsx

```diff
@@ -98,6 +114,66 @@ export default class CollectionsToolbar extends React.Component<Props, any> {
     else this.props.history.push(redirectURLS[button]);
   }
 
+  getCollectionId = () => {
+    const collectionPageResult = matchPath(this.props.location.pathname, {
+      path: COLLECTION_URL,
+      exact: true,
+    });
+    return get(collectionPageResult, "params.collection_id");
+  };
+
+  handleTitleChange = (e: any) => {
+    this.setState({ title: e.target.value });
+  };
+
+  handleTitleUpdate = async (e: any) => {
+    const collectionId = parseInt(this.getCollectionId());
+    let name = this.state.title;
+    let isPublic = this.props.isPublic;
+    if (collectionId) {
+      const resp = await renameCollection(
+        this.context,
+        collectionId,
+        name,
+        isPublic
+      );
+      console.log("rename response", resp);
+      await this.context.requestCollections({
+        limit: 100,
+        offset: 0,
+      });
+    }
+  };
+
+  submitOnEnter = async (e: any) => {
+    await this.handleTitleUpdate(e);
+    if (e.key === "Enter") {
+      this.setState({ isRenameMode: false, title: this.state.title });
+    }
+  };
+
+  submitOnClickOutside = async (e: any) => {
```

Comment:
I'm not sure we should be submitting when the user clicks outside. My first idea as a user, if I wanted to cancel the edit, would be to press ESC to cancel, and if that doesn't work I'd try clicking outside of the input.

Right now we offer no way of cancelling the edit